flow/compositor_context.h

```diff
@@ -18,72 +17,65 @@
 
 namespace flow {
 
+class LayerTree;
+
 class CompositorContext {
  public:
   class ScopedFrame {
    public:
+    ScopedFrame(CompositorContext& context,
+                GrContext* gr_context,
+                SkCanvas* canvas,
+                bool instrumentation_enabled);
+
+    virtual ~ScopedFrame();
+
     SkCanvas* canvas() { return canvas_; }
 
     CompositorContext& context() const { return context_; }
 
     GrContext* gr_context() const { return gr_context_; }
 
-    ScopedFrame(ScopedFrame&& frame);
-
-    ~ScopedFrame();
+    virtual bool Raster(LayerTree& layer_tree, bool ignore_raster_cache);
 
    private:
     CompositorContext& context_;
     GrContext* gr_context_;
     SkCanvas* canvas_;
     const bool instrumentation_enabled_;
 
-    ScopedFrame(CompositorContext& context,
-                GrContext* gr_context,
-                SkCanvas* canvas,
-                bool instrumentation_enabled);
-
-    friend class CompositorContext;
-
     FXL_DISALLOW_COPY_AND_ASSIGN(ScopedFrame);
   };
 
-  CompositorContext(std::unique_ptr<ProcessInfo> info);
+  CompositorContext();
 
-  ~CompositorContext();
+  virtual ~CompositorContext();
 
-  ScopedFrame AcquireFrame(GrContext* gr_context,
-                           SkCanvas* canvas,
-                           bool instrumentation_enabled = true);
+  virtual std::unique_ptr<ScopedFrame> AcquireFrame(
+      GrContext* gr_context,
+      SkCanvas* canvas,
+      bool instrumentation_enabled);
 
   void OnGrContextCreated();
 
   void OnGrContextDestroyed();
 
   RasterCache& raster_cache() { return raster_cache_; }
 
-  TextureRegistry& texture_registry() { return *texture_registry_; }
+  TextureRegistry& texture_registry() { return texture_registry_; }
 
   const Counter& frame_count() const { return frame_count_; }
 
   const Stopwatch& frame_time() const { return frame_time_; }
 
   Stopwatch& engine_time() { return engine_time_; }
 
-  const CounterValues& memory_usage() const { return memory_usage_; }
```

Comment:
Was wondering where this went and the answer is apparently we never used it. Nice! :)

Reply:
This was only ever wired up on iOS and never exposed in the framework.